### ISSUE TITLE
share, span options for twin x and y axes

### ISSUE THREAD

**Issue by kinyatoride** [lukelbd/proplot#205]
### Description

Hi, I really like the feature that `twinx` and `twiny` can be applied to `axs`, which was not possible with `matplotlib`.
Would it be possible to add `share` and `span` options to `twinx` and `twiny`? Such as `twinx(sharey=3, spany=True)`.
Currently, when we specified `ylabel` for twin axes, it will plot to all axes.

### Steps to reproduce

```python
import numpy as np
import proplot as plot

fig, axs = plot.subplots(nrows=2, ncols=2)
axs1 = axs.twinx()

x = np.arange(10)

for ax, ax1 in zip(axs, axs1):
    ax.plot(np.sin(x), c='b')
    ax1.plot(np.cos(x), c='r')
    
axs.format(ycolor='b', ylabel='ylabel1')
axs1.format(ycolor='r', ylabel='ylabel2')
```

**Current behavior**: 
ylabels and ticks are shown in every plots.
![Screen Shot 2020-07-10 at 10 45 00](https://user-images.githubusercontent.com/61028484/87183161-701e2f80-c29a-11ea-9c99-e7dbbcc5feb2.png)

### Proplot version

0.6.4

**Comment by lukelbd** [comment]
Yep this is a really important item on my to-do list -- you can monitor PR #183 for updates.
